modules/tinymce/src/core/main/ts/fmt/RemoveFormat.ts

```diff
@@ -517,6 +518,24 @@ const remove = (ed: Editor, name: string, vars?: FormatVars, node?: Node | Range
         }
       }
     }
+
+    // Note: Assists with cleaning up any stray text decorations that may been applied when text decorations
+    // and text colors were merged together from a applied format
+    // Remove child span if it only contains text-decoration and a parent node also has the same text decoration.
+    const textDecorations = [ 'underline', 'line-through', 'overline' ];
```

Comment:
I moved this, as it wasn't running in some cases when it really should have been from what I could tell.